src/effects/backends/builtin/echoeffect.cpp

```diff
@@ -236,16 +242,26 @@ void EchoEffect::processChannel(
             pGroupState->ping_pong = 0;
         }
     }
-
-    // The ramping of the send parameter handles ramping when enabling, so
-    // this effect must handle ramping to dry when disabling itself (instead
-    // of being handled by EngineEffect::process).
+    pGroupState->prev_send = send_current;
     if (enableState == EffectEnableState::Disabling) {
-        SampleUtil::applyRampingGain(pOutput, 1.0, 0.0, engineParameters.samplesPerBuffer());
-        pGroupState->delay_buf.clear();
+        // Function to determine if echo tail is gone
+        // Calculate absolute if the delayline-buffer is approx. zero/empty.
         pGroupState->prev_send = 0;
-    } else {
-        pGroupState->prev_send = send_current;
+        float differenceSum = 0.0f;
+        for (SINT i = 0; i < pGroupState->delay_buf.size();
+                i = i + 16) { // We don't need to consider *all* samples,
+                              // assuming 44.1kHz sample-rate this sums every
+                              // ~1/3ms
+            differenceSum += fabsf(pGroupState->delay_buf[i]);
+        }
+        // Calculate average of the differences
+        constexpr float threshold = 0.00001f;
+        const bool echoTailFullyFaded = differenceSum < threshold;
+        if (echoTailFullyFaded) {
+            m_isReadyForDisable = true;
+
+            pGroupState->delay_buf.clear();
+        }
```

Comment:
same here. lets just actually make a function.

Reply:
I made it into a function :--)